src/ListGroup.js

```diff
@@ -13,17 +13,15 @@ class ListGroup extends React.Component {
 
     if (!this.props.children) {
       return this.renderDiv(items);
-    } else if (React.Children.count(this.props.children) === 1 && !Array.isArray(this.props.children)) {
-      let child = this.props.children;
-
-      childrenAnchors = this.isAnchor(child.props);
-
     } else {
+      React.Children.forEach(this.props.children, (child) => {
+        if (!Array.isArray(child) && this.isAnchor(child.props)) {
+          childrenAnchors = true;
+        }
 
-      childrenAnchors = Array.prototype.some.call(this.props.children, (child) => {
-        return !Array.isArray(child) ? this.isAnchor(child.props) : Array.prototype.some.call(child, (subChild) => {
-            return this.isAnchor(subChild.props);
-        });
+        if (Array.isArray(child) && child.some.call((subChild) => this.isAnchor(subChild.props))) {
```

Comment:
In the general case probably not, but per https://facebook.github.io/react/docs/top-level-api.html#react.children.map:

>If `children` is a nested object or array it will be traversed: `fn` will never be passed the container objects.

And anyway the tests cover this and they still pass with the snippet I have above.